### ISSUE TITLE
Option to run mypy from arbitrary location

### ISSUE THREAD

**Issue by Hi-Angel** [python/mypy#9751]
**Feature**

A way to run mypy for project from arbitrary location is needed, such as: base directory option, or modifying behavior of `--config-file` option to use the directory with config as the base project directory. Or something else to that effect.

**Pitch**

I stumbled upon such problem twice in different settings for today, so I think it is very useful.

One usecase is: when you have one directory with python APIs, and you want build-system to run mypy on files in that directory. Right now you have to make the build system to go into that directory before running `mypy`, because otherwise, if you run for example `mypy --config-file=python-apis/.mypy.ini`, you'll get errors because mypy won't find files that are written in the ini file.

Another usecase: running mypy as part of lsp-server setup. For example, [`pyls-mypy` plugin](https://github.com/tomv564/pyls-mypy/) gets from pyls an absolute path to the file to run upon. Right now the file gets ultimately checked separately from the rest of the project, resulting in wrong errors:

```
Cannot find implementation or library stub for module named 'foo'
```

The module `foo` actually does exist, it's just in some of the parent directories, but there's no way to tell mypy that.

**Comment by hauntsaninja** [comment]
Does https://github.com/python/mypy/issues/7968 work for your use case?

**Comment by Hi-Angel** [comment]
> Does #7968 work for your use case?

So, I'm not completely sure if I'm using the env. variable correctly, but I get this:

```haskell
 λ MYPY_CONFIG_FILE_DIR=api/python/tests/ mypy --config-file=api/python/tests/.mypy.ini                                        
mypy: can't read file 'cleanup.py': No such file or directory
```

I tried without the `--config-file` option too, but `mypy` simply prints usage in this case, so apparently the option is necessary. So, unless I somehow misused the env. variable, it won't work for usecases discussed.

FTR, my mypy version is 0.790, which was released in October, so it should include the PR adding that env. variable, which was merged on September 4.

**Comment by Hi-Angel** [comment]
> FTR, my mypy version is 0.790, which was released in October, so it should include the PR adding that env. variable, which was merged on September 4.

Although, I just looked at the `config_parser.py` which was installed with `mypy`, and it doesn't include the change to parse the variable. Odd.

Either way, I added it manually *(it's [just two lines to copy-paste](https://github.com/python/mypy/pull/9414/files))*, and I can confirm that it does not change behavior compared to my prev. comment.

**Comment by Hi-Angel** [comment]
@hauntsaninja lol I actually just figured, the issue/PR linked do the opposite of what is discussed :-) What they do is that when `mypy` was run and found a config file, mypy would set environment variable with the directory of where that config file is located. That is, it's not something a user, such as a build system or an lsp-server, could set externally.

**Comment by hauntsaninja** [comment]
You're not supposed to set it, you're supposed to use it in your config file. That is, mypy sets the variable and uses it when parsing your config file, so if you e.g. have `files = $MYPY_CONFIG_FILE_DIR/cleanup.py` in your config file, it'll work.

**Comment by Hi-Angel** [comment]
> so if you e.g. have `files = $MYPY_CONFIG_FILE_DIR/cleanup.py` in your config file, it'll work

Oh, thank you, I see. Works for me this way. TBH can't say I like this solution, because it's a duplicated content before each file in the list, and the list as I understand can't be broken to multiple lines. So, as time goes, the line is gonna keep growing, and this variable is 22 additional characters to each new file. But in context of current feature-request it should work for usecases discussed, so I guess it can be closed.

Btw, for future readers: I cloned the repo locally, and can confirm that v0.790 release didn't include the changes. Should be in the next release.
